server/aesdsocket.c

```diff
@@ -0,0 +1,276 @@
+		syslog(LOG_ERR, "Couldn't get the server's address");
+		exit(5);
+	}
+	
+	//Bind
+	bind_status=bind(socket_fd,servinfo->ai_addr,sizeof(struct sockaddr));
+	if(bind_status==-1)
+	{
+		freeaddrinfo(servinfo); //Free the memory before exiting
+		syslog(LOG_ERR, "Binding failed");
+		exit(6);
+	}
+	
+	freeaddrinfo(servinfo); //Free this memory before next operations
+	
+	fd_status=open("/var/tmp/aesdsocketdata.txt", O_WRONLY | O_CREAT | O_TRUNC, S_IRWXU | S_IRWXG | S_IRWXO); //To create a file and give permissions to all
+	if (fd_status==-1)
+	{
+		syslog(LOG_ERR, "The file could not be created/found");
+		exit(7);
+	}
+	close(fd_status);
+	
+	//Infinite loop for server-client connections
+	while(1)
+	{
+	
+	//Listen
+	listen_status=listen(socket_fd,20); //Max queue of connections set to 20
+	if(listen_status==-1)
+	{
+		syslog(LOG_ERR, "Listening to the connections failed");
+		exit(8);
+	}
+	
+	//accept
+	accept_return=accept(socket_fd,(struct sockaddr *)&client_addr,&size);
+	if(accept_return==-1)
+	{
+		syslog(LOG_ERR, "Connection could not be accepted");
+		exit(9);
+	}
+	else
+	{
+		syslog(LOG_INFO,"Accepts connection from %s",inet_ntoa(client_addr.sin_addr));
+		printf("Accepts connection from %s\n",inet_ntoa(client_addr.sin_addr));
+	}
+	
+	
+	store_data = (char*)malloc(100); //Allocate a memory
+	if(store_data==NULL)
+	{
+		syslog(LOG_ERR, "Memory couldn't be allocated");
+		exit(10);
+	}
+	
+	memset(store_data,0,100); //Reset the temporary buffer
+	
+	write_flag=false;
+	
+	//Loop to receive data packet
+	while(!write_flag)
+	{
+	
+	rec_status=recv(accept_return,buff,100,0); //Receive data packets from client
+	if(rec_status==-1)
+	{
+		syslog(LOG_ERR, "Error in reception of data packets from client");
+		exit(11);
+	}
+	
+	
+	for(i=0;i<100;i++)
+	{
+		if(buff[i]=='\n')
+		{
+			i++;
+			write_flag=true;
+			break;
+		}	
+		
+	}
+	total_bytes+=i; //Total bytes in a data packet
+	store_data=(char *)realloc(store_data,total_bytes); //Realloc memory
+	if(store_data==NULL)
+	{
+		syslog(LOG_ERR, "Reallocation of memory failed");
+		exit(12);
+	}
+	memcpy(store_data+total_bytes-i,buff,i); //Copy data from temporary buffer to memory
+	memset(buff,0,100); //Reset the temporary buffer
+	}
+	
+	fd_status = open("/var/tmp/aesdsocketdata.txt", O_APPEND | O_WRONLY); //Open the file in writeonly and append mode
+	if(fd_status==-1)
+	{
+		syslog(LOG_ERR, "Could not open the file");
+		exit(13);
+	}
+		
+	
+	write_status= write(fd_status,store_data,total_bytes); //Writes to the file
+	if(write_status!=total_bytes)
+	{
+		syslog(LOG_ERR, "Could not write total bytes to the file");
+		exit(14);
+	}
+	close(fd_status);
+	
+	packet_bytes_total += total_bytes; //Total bytes of a connection
+		
+	int op_fd=open("/var/tmp/aesdsocketdata.txt",O_RDONLY); //Open file to read only
+	if(op_fd==-1)
+	{
+		syslog(LOG_ERR, "Could not open the file to read");
+		exit(15);
+	}
+		
+	char read_arr[packet_bytes_total]; //Create a temporary buffer to read file's contents
```

Comment:
@shashankchandrasekaran what assumption are you making here about the size of the stack and the total size of all packets sent?